lib/spack/spack/build_environment.py

```diff
@@ -563,46 +568,42 @@ def determine_number_of_jobs(
     return min(max_cpus, config_default)
 
 
-def set_module_variables_for_package(pkg):
+def set_module_variables_for_package(pkg, context="build"):
```

Comment:
Can we make `context` an `enum`?

Reply:
It's done everywhere except where it's assumed to be a string when serialized etc.